polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -185,12 +192,10 @@ pub struct PendingCollationFetch {
 pub enum CollationStatus {
 	/// We are waiting for a collation to be advertised to us.
 	Waiting,
-	/// We are currently fetching a collation.
-	Fetching,
-	/// We are waiting that a collation is being validated.
-	WaitingOnValidation,
-	/// We have seconded a collation.
-	Seconded,
+	/// We are currently fetching a collation for the specified `ParaId`.
+	Fetching(ParaId),
+	/// We are waiting that a collation is being validated for the specified `ParaId`.
+	WaitingOnValidation(ParaId),
 }
```

Comment:
Originally and before async backing once we've reached Seconded status we knew we are done here. Right now we can only be satisfied with collations once we reach the seconding limit but I see that we do not have a corresponding status here. 

When we reach the seconding limit and no more ocllations are expected what will be the status here? 

Or is it nonsensical to even refer to a CollationStatus when we reach the limit? (Dont think so)

Reply:
We could do it. The lifetime right now is Waiting -> Fetching -> WaitingOnValidation and back to Waiting. We can have an additional `Satisfied` or something similar which indicates that we can't accept any more collations at this relay parent. 

Then on each seconded collation we can call `seconded_and_pending_for_para_in_view` and if we have reached the claim queue limit we can set the state to `Satisfied` and know that nothing can be accepted at this relay parent.

The benefits are that once the relay parent 'is complete' (all claim queue entries are filled) we won't need to run any checks in order to reject an incoming collation.

The drawbacks are that on each seconded collation we'll have to run the same function. So unless a collator is spamming us the benefit is minimal. And since we are limiting the number of advertisements per collator we can't get spammed too much.

We can explore this further but it better be a follow up task.